### PR TITLE
For #23415 - Add support for GeckoView Screen Orientation APIs

### PR DESCRIPTION
**Pull request by Mugurell**
Needs https://github.com/mozilla-mobile/android-components/pull/11789
Blocked until https://bugzilla.mozilla.org/show_bug.cgi?id=1757431

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by Mugurell** [comment]
The flow used here is a bit cumbersome:
- register a middleware - which observes BrowserStore updates and informs us to execute `activity.setRequestedOrientation`
- send just the class of the activity which as a callback should set the new orientation - because `BrowserStore` has a longer lifespan than any activity and as such we can't pass an instance.
- receive the orientation callbacks through a new Intent from AC - reactive flow used for other features also which avoids a AC leaking certain Fenix app instances.

File: app/src/main/java/org/mozilla/fenix/home/intent/OrientationLockIntentProcessor.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+package org.mozilla.fenix.home.intent
+
+import android.content.Intent
+import android.content.pm.ActivityInfo
+import androidx.appcompat.app.AppCompatActivity
+import androidx.navigation.NavController
+import mozilla.components.browser.state.engine.middleware.EXTRA_REQUESTED_ORIENTATION
+
+class OrientationLockIntentProcessor(private val activity: AppCompatActivity) : HomeIntentProcessor {
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/home/intent/OrientationLockIntentProcessor.kt:13]: Drive by - Can you add a kDoc for what this class does

Comment (Mugurell) [reply app/src/main/java/org/mozilla/fenix/home/intent/OrientationLockIntentProcessor.kt:13]: Sure, documentation and tests to be added.

Just wanted first to validate the general flow which both here and in AC seems a bit too complex but it gets the job done while respecting the reactive arch and prevents uglier implementations with maybe a weak reference to activity and allows for communication of simple events not necessarily part of a MVI state.

**Review by jonalmeida — APPROVED** [review]
LGTM.

> Blocked until https://bugzilla.mozilla.org/show_bug.cgi?id=1757431

We also identified on the AC PR that this bug should not block integration.

**Pull request merged**